### ISSUE TITLE
Incorrect behaviour during routing with `$route`

### ISSUE THREAD

**Issue by prodigy7** [NickMcBurney/storybook-vue3-router#58]
As a team, we have now spent several hours fighting a bug in the library that has driven us to the edge of frustration.

Specifically, the issue is that we have a Vue component in which we access `$route`. If we call up the address `/app/products/123` for example, we get the correct path back directly via `$route.fullPath`. If the same component runs in Storybook with `initialRoute` set for named path, we only get `/` back here.
If we include `{{ $route.fullPath }}` in the template, we will first see `/` and shortly afterwards it will switch to the route that was actually called.

If we then make changes so that there is a hot reload, the correct value is also returned on the console via `console.log(this.$route.fullPath)` in the mounted method. This means that something must go wrong during initialisation so that the component is rendered before the routing has been completed.

Version installed of the add-on show by `npm ls` is 4.25.0 (a bit strange, i see only 4.1.0 here https://github.com/NickMcBurney/storybook-vue3-router/releases).

**Comment by NickMcBurney** [comment]
@prodigy7, sorry your having trouble. Can you provide a reproduction and show the plugin set-up your using?

I'm not actually 100% certain what the issue is your having but it sounds like:
- initialRoute setting not setting $route as expected (as it appears to only correctly set after component is mounted and not when router is initialised).

P.s. my release pipeline failed and this caused it to release versions 4.1.0-4.25.0 at the same time. Due to a other package which has this packaged installed with a weak version specification I can't delete the duplicate versions from npm.

**Comment by prodigy7** [comment]
I've requested a clean feature for allow override the url here: https://github.com/vuejs/router/issues/2063

**Comment by prodigy7** [comment]
Okay, I've found a workaround for now:

```javascript
@@ -1,5 +1,5 @@
 <template>
-  <q-page class="q-pa-md full-width row items-start">
+  <q-page class="q-pa-md full-width row items-start" v-if="vueRouterReady">
     <div class="col-lg-4 col-xm-12 col-xs-12" id="contentPanelHeader">
       <div class="q-px-md items-start">
         <div class="row">
@@ -116,7 +116,11 @@ export default {
       }).Bookings();
     }
   },
-  computed: {},
+  computed: {
+    vueRouterReady() {
+      return(this.$route.path != '/');
```

Means: If the route isn't set to the expected, do not render the component. When the route is set, render. In my case, the component uses computed values which access `$route`. Because `$route` does not contain the needed parameters, my code fails. Now, the question would be: Is it maybe possible inside the add on prevent rendering the component so long the route does not match the expected initialRoute?

**Comment by prodigy7** [comment]
Okay, found a second more elegant solution:

```javascript
import Details from './Details.vue';
import { transformVueComponentPropListToArgTypes } from "@/../.storybook/storybookHelpers.js"

import routes from '@/router/routes.js';
import {vueRouter} from 'storybook-vue3-router'

export default {
  component: Details,
  tags: ['autodocs'],
  argTypes: transformVueComponentPropListToArgTypes(Details.props),
  decorators: [
    vueRouter(routes, {
      initialRoute: '/app/products/3974'
    })
  ]
};

const Template = (args) => ({
  components: {
    Details
  },
  template: `
    <q-layout v-if="$route.path != '/'"><q-page-container><Details /></q-page-container></q-layout>
  `
})

export const Default = Template.bind({});
Default.args = {
}
```

In this case the story starts to render if the condition matches, that the route isn't `/`. The question now would simply be whether it is possible to link into the rendering globally in such a way that I no longer have to set the condition.

**Comment by NickMcBurney** [comment]
Hi @prodigy7, so I've got a version working locally which makes use of vue router's `await router.isReady()` in order to ensure router is setup and `initialRoute` is set before rendering the story. 

It works, but in order to use this I have to make the storybook decorator return an `async setup()` - This means you then need to update `.storybook/preview.ts` to wrap components in Vue's `<Suspense>` component - which allows for async setup.

Your `preview.ts` would need this adding:

![image](https://github.com/NickMcBurney/storybook-vue3-router/assets/6604187/c504d3b2-9e42-4f70-a599-7cd517cdcc3e)

TBH I'm not sure if this is a better solution than what you have which only renders router dependant component once route has changed to `initialRoute` value.

^^ It will also require me to provide a new exported method in this package to specifically define the `async` + `await router.isReady()` method in order to not break setup for other users.

**Comment by NickMcBurney** [comment]
@prodigy7 I've published a new version 5.0.0, this includes a new `asyncVueRouter` export which will work as mentioned above.

For your use-case it may be easier to keep your current setup. But if you want you can use new decorator (I suspect there is other use-cases which might require this). 

I've [updated docs](https://github.com/NickMcBurney/storybook-vue3-router#routerisready), and you can see this commit which shows changes required in preview.js and a story example https://github.com/NickMcBurney/storybook-vue3-router/commit/db4eda509f064e2c635ac5e198678bfc9514fb4b

**Comment by prodigy7** [comment]
@NickMcBurney Sorry for my late reply. Thank you very much, it works how we need it :-)

Perhaps just one more question: Is it possible to extend it so that the loading display of Storybook appears until the story is also displayed? We have components that sometimes take a little longer to load and in such a situation you see a blank white page for so long.
